packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -1,273 +1,69 @@
 import * as React from 'react';
-import PropTypes from 'prop-types';
-import { chainPropTypes } from '@material-ui/utils';
-import { GridComponent, GridComponentProps, GridRowId, useThemeProps } from '../../_modules_/grid';
+import { DATAGRID_PROPTYPES } from './DataGridPropTypes';
+import {
+  DEFAULT_GRID_OPTIONS,
+  GridBody,
+  GridComponentProps,
+  GridErrorHandler,
+  GridFooterPlaceholder,
+  GridHeaderPlaceholder,
+  GridRoot,
+  useGridApiRef,
+} from '../../_modules_/grid';
+import { GridContextProvider } from '../../_modules_/grid/context/GridContextProvider';
+import { useDataGridComponent } from './useDataGridComponent';
+import { DataGridProps, MAX_PAGE_SIZE } from './DataGridProps';
 
-const FORCED_PROPS: Partial<GridComponentProps> = {
+const DATA_GRID_FORCED_PROPS: Omit<GridComponentProps, keyof DataGridProps> = {
+  apiRef: undefined,
   disableColumnResize: true,
   disableColumnReorder: true,
   disableMultipleColumnsFiltering: true,
   disableMultipleColumnsSorting: true,
   disableMultipleSelection: true,
   pagination: true,
-  apiRef: undefined,
   onRowsScrollEnd: undefined,
   checkboxSelectionVisibleOnly: false,
 };
 
-export type DataGridProps = Omit<
-  GridComponentProps,
-  | 'apiRef'
-  | 'checkboxSelectionVisibleOnly'
-  | 'disableColumnResize'
-  | 'disableColumnReorder'
-  | 'disableMultipleColumnsFiltering'
-  | 'disableMultipleColumnsSorting'
-  | 'disableMultipleSelection'
-  | 'licenseStatus'
-  | 'options'
-  | 'onRowsScrollEnd'
-  | 'pagination'
-  | 'scrollEndThreshold'
-  | 'selectionModel'
-> & {
-  apiRef?: undefined;
-  checkboxSelectionVisibleOnly?: false;
-  disableColumnResize?: true;
-  disableColumnReorder?: true;
-  disableMultipleColumnsFiltering?: true;
-  disableMultipleColumnsSorting?: true;
-  disableMultipleSelection?: true;
-  onRowsScrollEnd?: undefined;
-  selectionModel?: GridRowId | GridRowId[];
-  pagination?: true;
-};
-
-const MAX_PAGE_SIZE = 100;
+export const DATA_GRID_DEFAULT_PROPS = DEFAULT_GRID_OPTIONS;
```

Comment:
Is there a reason to export `DATA_GRID_DEFAULT_PROPS` and `X_GRID_DEFAULT_PROPS`?

Reply:
https://github.com/mui-org/material-ui-x/pull/2176#discussion_r677743672

To use them here
But if we stop using them in test I can stop exporting them